src/python/grpcio/commands.py

```diff
@@ -211,6 +212,20 @@ class BuildExt(build_ext.build_ext):
     }
     LINK_OPTIONS = {}
 
+    def get_ext_filename(self, ext_name):
+      # since python3.5, python extensions' shared libraries use a suffix that corresponds to the value
+      # of sysconfig.get_config_var('EXT_SUFFIX') and contains info about the architecture the library targets.
+      # E.g. on x64 linux the suffix is ".cpython-XYZ-x86_64-linux-gnu.so"
+      # When crosscompiling python wheels, we need to be able to override this suffix
+      # so that the resulting file name matches the target architecture and we end up with a well-formed
+      # wheel.
```

Comment:
Does setuptool try to find compilers based on the file name? Good finding.

Reply:
So when the python extension attempts to be loaded, the filename suffix needs to match the configuration of the python you're running.  The full docs is here: https://www.python.org/dev/peps/pep-3149/